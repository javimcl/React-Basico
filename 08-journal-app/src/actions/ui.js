import { type } from "@testing-library/user-event/dist/type";
import { types } from "../types/types"


export const setError = (err) => ({
    type: types.uiSetError,
    payload: err
});

export const removeError = () => ({
    type: types.uiRemoverError,
})

export const startLoading = () => ({
    type: types.uiStartLoading
})

export const finishLoading  = () => ({
    type: types.uiFinishLoading
})
