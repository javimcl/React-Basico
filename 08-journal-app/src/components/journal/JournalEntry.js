import React from 'react';

export const JournalEntry = () => {
  return (
      <div className='journal__entry pointer'>
          <div 
          className='journal__entry-picture'
          style={{
              backgroundSize: 'cover',
              backgroundImage: 'url(https://earthsky.org/upl/gravity_forms/6-d184048789a60d766a02a8e43117298b/2022/02/20220130-Rosette-Nebula-STO.jpg)'
          }}
          >

          </div>
          <div className='journal__entry-body'>
              <p className='journal__entry-title'>
                  Un nuevo dia
              </p>
              <p className='journal__entry-content'>
                  Reprehenderit in duis
              </p>


          </div>
          <div className='journal__entry-date-box'>
              <span>Monday</span>
              <h4>28</h4>

          </div>
      </div>
  );
};
