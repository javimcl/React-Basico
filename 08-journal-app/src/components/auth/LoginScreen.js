import React from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { login } from '../../actions/auth';
import { useForm } from '../../hooks/useForm';

export const LoginScreen = () => {

  const dispatch = useDispatch();

  const [formValues, handleInputChange] = useForm({
    email: 'javier',
    password: '1234'
  });

  const { email, password } = formValues;

  const handleLogin = (e) => {
    e.preventDefault();
    dispatch(login(123,'Javier'))

  }


  return (
    <>
      <h3 className="auth__title">Login</h3>
      <form onSubmit={handleLogin}>
        <input
          type="text"
          placeholder="Email"
          name="email"
          className="auth__input"
          autoComplete='off'
          value={email}
          onChange={handleInputChange}
        />
        <input
          type="password"
          placeholder="Password"
          name="password"
          className="auth__input"
          value={password}
          onChange={handleInputChange}
        />
        <button className='btn btn-primary btn-block'
          type="submit"
        >
          Login
        </button>

        <div className='auth__social-networks'>
          <p>Login with social metworks</p><div
            className="google-btn"
          >
            <div className="google-icon-wrapper">
              <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
            </div>
            <p className="btn-text">
              <b>Sign in with google</b>
            </p>
          </div>
        </div>
        <Link className='link'
          to="/auth/register">
          Create new account
        </Link>

      </form>
    </>
  );
};
