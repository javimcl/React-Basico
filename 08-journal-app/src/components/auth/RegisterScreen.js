import React from 'react';
import { Link } from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import validator from 'validator';

import { setError, removeError } from '../../actions/ui'
import { useDispatch, useSelector } from 'react-redux';
import { startRegisterWithEmailPasswordName } from '../../actions/auth';

export const RegisterScreen = () => {

  const dispatch = useDispatch();

  const { msgError } = useSelector(state => state.ui);

 // console.log(msgError);

  const initial = {
    name: 'Hernando',
    email: 'nando@iess.gob.ec',
    password: '123456',
    password2: '123456'
  }

  const [formValues, handleInputChange] = useForm(initial);

  const { name, email, password, password2 } = formValues;

  const handleRegister = (e) => {
    e.preventDefault();    
    if (isFormValid()) {
      dispatch(startRegisterWithEmailPasswordName(email, password, name))
    }
  }

  const isFormValid = () => {

    if (name.trim().length === 0) {
      dispatch(setError('Name is required'));
      return false;
    } else if (!validator.isEmail(email)) {      
      dispatch(setError('Email is not valida'));
      return false;

    } else if (password !== password2 || password.length < 5) {
      dispatch(setError('Password should be at least 6 characters and match each other'))
      return false;
    }

    dispatch(removeError());


    return true;
  }


  return (
    <>
      <h3 className="auth__title">Register</h3>
      <form onSubmit={handleRegister}
        className='animate__animated animate__fadeIn animate__faster'>
        {
          msgError &&
          (
            <div className='auth__alert-error'>
              {msgError}
            </div>
          )
        }
        <input
          type="text"
          placeholder="Name"
          name="name"
          className="auth__input"
          autoComplete='off'
          onChange={handleInputChange}
          value={name}
        />
        <input
          type="text"
          placeholder="Email"
          name="email"
          className="auth__input"
          autoComplete='off'
          onChange={handleInputChange}
          value={email}
        />
        <input
          type="password"
          placeholder="Password"
          name="password"
          className="auth__input"
          onChange={handleInputChange}
          value={password}
        />
        <input
          type="password"
          placeholder="Confirm Password"
          name="password2"
          className="auth__input"
          onChange={handleInputChange}
          value={password2}
        />
        <button className='btn btn-primary btn-block mb-5'
          type="submit"
          onClick={handleRegister}
        >
          Register
        </button>

        <div className='auth__social-networks'>
          <p>Login with social metworks</p><div
            className="google-btn"
          >
            <div className="google-icon-wrapper">
              <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
            </div>
            <p className="btn-text">
              <b>Sign in with google</b>
            </p>
          </div>
        </div>
        <Link className='link'
          to="/auth/login">
          Already registered?
        </Link>

      </form>
    </>
  );
};
