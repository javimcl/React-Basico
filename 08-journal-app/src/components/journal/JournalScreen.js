import React from 'react';
import { NodeScreen } from '../notes/NodeScreen';
import { NothingSelected } from './NothingSelected';
import { Sidebar } from './Sidebar';

export const JournalScreen = () => {
  return (
      <div className="journal__main-content">
          <Sidebar/>

          <main>
            {/* <NothingSelected/> */}
            <NodeScreen/>
          </main>
      </div>
  );
};
