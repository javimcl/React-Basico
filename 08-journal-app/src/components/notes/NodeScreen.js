import React from 'react';
import { NotesAppBar } from './NotesAppBar';

export const NodeScreen = () => {
    return (
        <div className='notes__main-content'>
            <NotesAppBar />
            <div className='notes__content'>
                <input
                    type='text'
                    placeholder='Some awesome title'
                    className='noter__title-input'
                    autoComplete='off'>
                </input>
                <textarea
                    placeholder='What happened today'
                    className='notes__textarea'>

                </textarea>
                <div className='notes__image'>
                    <img
                        src='https://earthsky.org/upl/gravity_forms/6-d184048789a60d766a02a8e43117298b/2022/02/DSC_2079.JPG'
                        alt='imagen'
                    />

                </div>

            </div>

        </div>
    );
};
