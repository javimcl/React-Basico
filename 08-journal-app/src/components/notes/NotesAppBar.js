import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNote } from '../../actions/notes';

export const NotesAppBar = () => {

    const dispatch = useDispatch();
    const { active } = useSelector(state => state.notes);
    const handleSave = () => {
        dispatch(startSaveNote(active));
    }


    const handlePictureClick = () => {

    }

    const handleFileChange = () => {
        
    }

    return (
        <div className='notes__appbar'>
            <span>28 de agosto 2022</span>

            <input 
            type="file"
            style={{ display: 'none'}}
            onChange={handleFileChange}>
            </input>






            <div>
                <button className='btn'
                onClick={handlePictureClick}>
                    Picture
                </button>
                <button className='btn'
                    onClick={handleSave}>
                    Save
                </button>
            </div>

        </div>
    );
};
