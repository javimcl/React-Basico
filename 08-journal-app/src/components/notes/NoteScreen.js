import React, { useEffect, useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { activeNote } from '../../actions/notes';
import { useForm } from '../../hooks/useForm';
import { NotesAppBar } from './NotesAppBar';

export const NoteScreen = () => {

    const dispatch = useDispatch();
    const { active: note } = useSelector(state => state.notes);

    const [formValues, handleInputChange, reset] = useForm(note);
    const { body, title } = formValues;

    const activeId = useRef(note.id);


    useEffect(() => {
        if (note.id !== activeId.current) {
            reset(note);
            activeId.current = note.id;
        }
    }, [note, reset])

    useEffect(() => {
        dispatch(activeNote(formValues.id, {...formValues}))
      
    }, [formValues, dispatch])
    


    return (
        <div className='notes__main-content'>
            <NotesAppBar />
            <div className='notes__content'>
                <input
                    type='text'
                    placeholder='Some awesome title'
                    className='noter__title-input'
                    value={title}
                    name="title"
                    autoComplete='off'
                    onChange={handleInputChange}>
                </input>
                <textarea
                    placeholder='What happened today'
                    className='notes__textarea'
                    value={body}
                    name='body'
                    onChange={handleInputChange}
                >

                </textarea>

                {
                    (note.url)
                    &&
                    (<div className='notes__image'>
                        <img
                            src='https://earthsky.org/upl/gravity_forms/6-d184048789a60d766a02a8e43117298b/2022/02/DSC_2079.JPG'
                            alt='imagen'
                        />

                    </div>)
                }

            </div>

        </div>
    );
};
