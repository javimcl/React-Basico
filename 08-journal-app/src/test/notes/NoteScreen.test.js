import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';

import '@testing-library/jest-dom';

import { mount } from 'enzyme'
import React from 'react'
import { Provider } from "react-redux";
import { Sidebar } from '../../components/journal/Sidebar';
import { startLogout } from '../../actions/auth';
import { activeNote, startNewNote } from '../../actions/notes';
import { NoteScreen } from '../../components/notes/NoteScreen';

jest.mock('../../actions/notes', () => ({
    activeNote: jest.fn()

}))


const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {
        uid: '1',
        name: 'Javier'
    },
    ui: {
        loading: false,
        msgError: null
    },
    notes: {
        active: {
            id: '1234',
            title: 'title',
            body: 'body',
            date: 0
        },
        notes: [],
    }
};

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <NoteScreen />
        </MemoryRouter>
    </Provider>
);

describe('Pruebas en <Sidebar/>', () => {
    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();

    })

    test('debe de disparar el active note', () => {

        // console.log(wrapper.find('button').exists);
        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title',
                value: 'Hola de nuevo'
            }
        });
        expect(activeNote).toHaveBeenLastCalledWith(
            "1234",
            {
                id: "1234",
                title: 'Hola de nuevo',
                body: 'body',
                date: 0

            }
        );
    })

})