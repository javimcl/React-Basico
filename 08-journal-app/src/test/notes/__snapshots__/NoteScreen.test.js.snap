// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Sidebar/> debe de mostrarse correctamente 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      28 de agosto 2022
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="noter__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="title"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today"
      value="body"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
